frontend/src/models/announcements.ts

```diff
@@ -0,0 +1,119 @@
+import { createModel } from '@rematch/core'
+import { ApplicationState } from '../store'
+import { graphQLRequest, graphQLGetErrors, graphQLCatchError } from '../services/graphQL'
+import { graphQLReadNotice } from '../services/graphQLMutation'
+import { AxiosResponse } from 'axios'
+import { RootModel } from './rootModel'
+
+type IAnnouncementsState = ILookup<IAnnouncement[]> & {
+  all: IAnnouncement[]
+}
+
+const announcementState: IAnnouncementsState = {
+  all: [],
+}
+
+export default createModel<RootModel>()({
+  state: announcementState,
+  effects: dispatch => ({
+    async fetch() {
+      try {
+        const result = await graphQLRequest(
+          ` {
+            notices {
+              id
+              title
+              body
+              image
+              link
+              type
+              modified
+              read
+            }
+          }`
+        )
+        const all = await dispatch.announcements.parse(result)
+        dispatch.announcements.set({ all })
+        graphQLGetErrors(result)
+      } catch (error) {
+        await graphQLCatchError(error)
+      }
+    },
+    async parse(result: AxiosResponse<any> | undefined): Promise<IAnnouncement[]> {
+      const all = result?.data?.data?.notices
+      // const all = TEST_DATA
+      console.log('NOTICES', all)
+      return all.map(n => ({
+        id: n.id,
+        title: n.title,
+        body: n.body,
+        image: n.image,
+        link: n.link,
+        type: n.type,
+        modified: new Date(n.modified),
+        read: n.read ? new Date(n.read) : undefined,
+      }))
+    },
+    async read(id: string, state) {
+      console.log('ANNOUNCEMENT READ', id)
+      await graphQLReadNotice(id)
+      dispatch.announcements.setRead(id)
+    },
+  }),
+  reducers: {
+    setRead(state, id: string) {
+      state.all.find((a, i) => {
+        if (a.id === id) state.all[i].read = new Date()
+      })
+      return state
+    },
+    set(state, params: ILookup<IAnnouncement[]>) {
+      Object.keys(params).forEach(key => (state[key] = params[key]))
+      return state
+    },
+  },
+})
+
+export function selectAnnouncements(state: ApplicationState, unread?: boolean) {
+  return state.announcements.all.filter(a => !unread || !a.read)
+}
+
+const TEST_DATA: any = [
```

Comment:
do you need this TEST_DATA?

Reply:
Don't need, but it's useful for testing.